src/torchcodec/_core/CudaDevice.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include "src/torchcodec/_core/DeviceInterface.h"
+
+namespace facebook::torchcodec {
+
+class CudaDevice : public DeviceInterface {
```

Comment:
The term "device" should be reserved for the concept of pytorch device object. Here, the `CudaDevice` class name is misleading because it doesn't refer to a device, it refers to a device **interface**. @dvrogozh , do you mind submitting a PR to rename CudaDevice into `CudaDeviceInterface` (file names and file classes)?

Reply:
@NicolasHug : no problem, will do.

Reply:
@NicolasHug : filed #626